addons/mail/models/mail_alias.py

```diff
@@ -206,13 +207,22 @@ def get_alias_values(self):
     @api.model
     def create(self, vals):
         """ Create a record with ``vals``, and create a corresponding alias. """
+        if 'incoming_email' in vals and not vals.get('incoming_email'):
+            vals['alias_name'] = False
         record = super(AliasMixin, self.with_context(
             alias_model_name=self.get_alias_model_name(vals),
             alias_parent_model_name=self._name,
         )).create(vals)
         record.alias_id.sudo().write(record.get_alias_values())
         return record
 
+    @api.multi
+    def write(self, vals):
```

Comment:
Please remove that kind of code. Aliases don't have to be managed with magic fields like that.